alg/cuCompactor.cuh

```diff
@@ -0,0 +1,165 @@
+/*
+ * cuCompactor.h
+ *
+ *  Created on: 21/mag/2015
+ *      Author: knotman
+ */
+
+#ifndef CUCOMPACTOR_H_
+#define CUCOMPACTOR_H_
+
+#include <thrust/scan.h>
+#include <thrust/device_vector.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+// Define this to turn on error checking
+#define CUDA_ERROR_CHECK
+
+#define CUDASAFECALL( err ) __cudaSafeCall( err, __FILE__, __LINE__ )
+
+inline void __cudaSafeCall( cudaError err, const char *file, const int line )
+{
+#ifdef CUDA_ERROR_CHECK
+	if ( cudaSuccess != err )
+	{
+		fprintf( stderr, "cudaSafeCall() failed at %s:%i : %s\n",
+				file, line, cudaGetErrorString( err ) );
+
+		fprintf( stdout, "cudaSafeCall() failed at %s:%i : %s\n",
+						file, line, cudaGetErrorString( err ) );
+		exit( -1 );
+	}
+#endif
+
+	return;
+}
+
+namespace cuCompactor {
+
+#define warpSize (32)
+#define FULL_MASK 0xffffffff
+
+__host__ __device__ int divup(int x, int y) { return x / y + (x % y ? 1 : 0); }
+
+__device__ __inline__ int pow2i (int e){
+	return 1<<e;
+}
+
+
+template <typename T,typename Predicate>
+__global__ void computeBlockCounts(T* d_input,int length,int*d_BlockCounts,Predicate predicate){
```

Comment:
Could you please add some comments above each of these functions to briefly describe what the functions do?

Reply:
We are no longer using these functions.